src/main/java/bral/projectile/ProjectileFrame.java

```diff
@@ -0,0 +1,140 @@
+        velocityField.getDocument().addDocumentListener(new SimpleDocumentListener() {
+            @Override
+            public void update(DocumentEvent e) {
+                recalculateAndDisplay(angleSlider, velocityField, secondsField, xResultLabel, yResultLabel, peakYResultLabel, interceptXResultLabel);
+            }
+        });
+
+        secondsField.getDocument().addDocumentListener(new SimpleDocumentListener() {
+            @Override
+            public void update(DocumentEvent e) {
+                recalculateAndDisplay(angleSlider, velocityField, secondsField, xResultLabel, yResultLabel, peakYResultLabel, interceptXResultLabel);
+            }
+        });
+        angleSlider.addChangeListener(new ChangeListener() {
+            @Override
+            public void stateChanged(ChangeEvent e) {
+                recalculateAndDisplay(angleSlider, velocityField, secondsField, xResultLabel, yResultLabel, peakYResultLabel, interceptXResultLabel);
+            }
+        });
+
+        // a listener that does something when the button is clicked
+        calculateButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                // create a new projectile
+                Projectile projectile = new Projectile(
+                        angleSlider.getValue(),
+                        Double.parseDouble(velocityField.getText())
+                );
+                projectile.setSeconds(
+                        Double.parseDouble(secondsField.getText())
+                );
+                // display info
+                xResultLabel.setText(Double.toString(projectile.getX()));
+                yResultLabel.setText(Double.toString(projectile.getY()));
+                peakYResultLabel.setText(Double.toString(projectile.getPeakY()));
+                interceptXResultLabel.setText(Double.toString(projectile.getInterceptX()));
+
+            }
+        });
+
+    }
+
+    private void recalculateAndDisplay(
+            JSlider angleSlider, JTextField velocityField, JTextField secondsField,
+            JLabel xResultLabel, JLabel yResultLabel, JLabel peakYResultLabel, JLabel interceptXResultLabel
+            )
+    {
+        try
+        {
+            Projectile projectile = new Projectile(
+                    angleSlider.getValue(),
+                    Double.parseDouble(velocityField.getText())
+            );
+            projectile.setSeconds(
+                    Double.parseDouble(secondsField.getText())
+            );
+            // display info
+            xResultLabel.setText(Double.toString(projectile.getX()));
+            yResultLabel.setText(Double.toString(projectile.getY()));
+            peakYResultLabel.setText(Double.toString(projectile.getPeakY()));
+            interceptXResultLabel.setText(Double.toString(projectile.getInterceptX()));
+        }
+        catch (NumberFormatException e)
+        {
+            e.getMessage();
```

Comment:
This doesn't do anything.
What do you want to do?

Reply:
It throws an error if not all the fields are filled in yet (when the user enters the numbers for the first time)

Reply:
I mean the specific line
e getMessage()
Doesn't do anything